src/EditorFeatures/Core/Implementation/MetadataAsSource/MetadataAsSourceFileService.cs

```diff
@@ -224,10 +225,13 @@ private string GetRootPathWithGuid_NoLock()
             }
 
             // Load the assembly.
-            var assemblyDefinition = AssemblyDefinition.ReadAssembly(assemblyLocation, new ReaderParameters() { AssemblyResolver = new RoslynAssemblyResolver(compilation) });
+            // TODO: Use a different PEFile ctor overload that allows us to reuse the PEReader or Stream already created by Roslyn.
```

Comment:
❗️  These comments are best removed before merge, and follow-up issues filed for them. The follow-up issues can be created before or after merging the pull request, but when you remove the TODO comments you should either create the issues directly or add comments to the pull request with information necessary to create the issues (so we don't lose the info). 😄

Reply:
I think we will have to always load the assembly being decompiled from disk, because will load them in "metadata-only mode" I guess.

As for the other references, could you tell me how to access the PEReader or Stream used by Roslyn? Thanks!
I would be glad to implement this correctly right from the start.